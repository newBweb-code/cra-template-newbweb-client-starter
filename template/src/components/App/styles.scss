  .root {
    width: '100%';
    min-width: 600px;
    font-size:1.5rem;
    overflow: auto;

    @media (max-width: 600px) {
      font-size: 1rem;
    }
  }

  .header,
  .footer {
    height: 100px;
    line-height: 100px;
    text-align: center;
    background-color: rgba(0,0,0,.8);
    color: #fff;
    overflow: hidden;
  }

  .main {
    display:flex;
    padding: 48px;
    min-height: calc(100vh - 200px);
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    text-align: center;
  }

  .logo {
    height: 100px;
    padding:2px;
  }

  .code {
    margin: 16px;
    padding: 32px 64px;
    border-radius: 8px;
    background-color: #292d3e;
    color: #bfc7d5;
    font-family: "Roboto Mono";
    text-align: left;
    word-wrap: none;
    animation: pulse 5s infinite;
  }

@keyframes pulse {
  0% { box-shadow: 0 0 4px 4px transparent; }
  30% { box-shadow: 0 0 4px 4px #f00; }
  40% { box-shadow: 0 0 4px 4px #f00; }
  90% { box-shadow: 0 0 4px 4px transparent; }
}
