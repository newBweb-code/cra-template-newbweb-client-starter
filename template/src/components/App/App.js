import React from 'react';
import ExternalLink from 'components/ExternalLink';
import './styles.scss';
import logo from 'assets/images/newbweb-logo-512-white.png';

function App() {
  return (
    <div className="root">
      <header className="header"><img src={logo} alt="newBweb" className="logo" /></header>
      <main className="main">
        <h1>React Client Starter</h1>
        <p>This template was built to follow along with newBweb's "Serverless React for Newbies" course.</p>
        <p>
          Make sure you ran the following command after using Create React App,
          or you will be missing some dev dependencies!
        </p>
        <code className="code">
          node addDevDependencies.js
          <br />
          npm install
        </code>
      </main>
      <footer className="footer">
        Visit
        {' '}
        <ExternalLink href="https://newbweb.com">newBweb website</ExternalLink>
        {' '}
        for more info!
      </footer>
    </div>
  );
}

export default App;
