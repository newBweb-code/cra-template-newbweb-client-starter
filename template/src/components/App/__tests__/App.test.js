import React from 'react';
import { render } from 'enzyme';

import App from '../App';

describe('<App />', () => {
  it('to match snapshot.', () => {
    // When
    const snapshot = render(<App />);
    // Then
    expect(snapshot).toMatchSnapshot();
  });
});
