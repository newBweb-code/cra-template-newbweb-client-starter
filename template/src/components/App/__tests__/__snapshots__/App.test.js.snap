// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> to match snapshot. 1`] = `
<div
  class="root"
>
  <header
    class="header"
  >
    <img
      alt="newBweb"
      class="logo"
      src="newbweb-logo-512-white.png"
    />
  </header>
  <main
    class="main"
  >
    <h1>
      React Client Starter
    </h1>
    <p>
      This template was built to follow along with newBweb's "Serverless React for Newbies" course.
    </p>
    <p>
      Make sure you ran the following command after using Create React App, or you will be missing some dev dependencies!
    </p>
    <code
      class="code"
    >
      node addDevDependencies.js
      <br />
      npm install
    </code>
  </main>
  <footer
    class="footer"
  >
    Visit 
    <a
      href="https://newbweb.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      newBweb website
    </a>
     for more info!
  </footer>
</div>
`;
